### ISSUE TITLE
Open-XML-SDK.sln fails to build due to incorrect ProjectLoadStyle environment variable

### ISSUE THREAD

**Issue by ThomasBarnekow** [OfficeDev/Open-XML-SDK#652]
**Description**

The Open-XML-SDK.sln fails to build in both Visual Studio 2019 Version 16.4.2 and on the command line.

In Visual Studio, using the DocumentFormat.OpenXml.csproj as an example, the error message for all projects is:
```
Error    NETSDK1013    The TargetFramework value '' was not recognized. It may be misspelled. If not, then the TargetFrameworkIdentifier and/or TargetFrameworkVersion properties must be specified explicitly.    DocumentFormat.OpenXml    C:\Program Files\dotnet\sdk\3.1.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets    93
```

On the command line, the error message shown by `dotnet restore` is:

```
C:\Program Files\dotnet\sdk\3.1.100\NuGet.targets(123,5): error : Value cannot be null. (Parameter 'folderName') [[local path removed]\Open-XML-SDK\Open-XML-SDK.sln]
```

**Information**

- Visual Studio: Community 2019 Version 16.4.2 (with previous versions, the projects can't even be opened)
- .NET Target: Any (did not explicitly specify)
- DocumentFormat.OpenXml Version: commit 871600ce22399c69bbbbb4c7f2eed93a59aa580b on master
- `ProjectLoadStyle` environment variable set to `DevCore10` or `DevCore31`

**Repro**

1. Create a fresh clone of the Open XML SDK, using `git clone https://github.com/OfficeDev/Open-XML-SDK.git`
2. Build using Visual Studio or on the command line:
    - In Visual Studio: open and build the solution.
    - On the command line: go to the top-level folder with `.sln` file and execute `dotnet restore`

**Observed**

As described above, Visual Studio shows build errors and `dotnet restore` prints an error message.

**Expected**

It should be possible to build the latest commit on master. The build instructions should include important details (e.g., environment variables). Where environment variables, for example, have become obsolete, a default configuration should be built.

**Comment by twsouthwick** [comment]
Hmm interesting... I can't repro this. It does require .NET Core SDK to build (as it by default will target .NET Standard 2.0). Do you have that installed?

**Comment by rominator1983** [comment]
Cant repro on my side too. Everything building fine. I have VS professional (also 16.4.2) and had .NET Core SDK 3.1 installed too.
Some colleagues reported that the .NET Core 2.1 that is shipped with VS has some kind of quirks and produced strange build errors. Installing .NET Core 3.1 (non preview) fixed the issue along with cleanly removing the 3.1 preview.

**Comment by ThomasBarnekow** [comment]
I think I found the culprit. I still had the `ProjectLoadStyle` environment variable set to `DevCore10`. I then looked at the `Directory.Build.props` file and saw the following:

```
    <!-- Because of the size of the project, to facilitate  quick development, by default only single
         frameworks will be build. This is customizable with the following possible values:

         - DevFramework40: .NET Framework 4.0
         - DevFramework46: .NET Framework 4.6
         - DevCore11:      .NET Core 1.1
         - DevCore21:      .NET Core 2.1
         - DevCore31:      .NET Core 3.1
         - All: Will build for all platforms
         -->
```

So it seems DevCore10 is no longer used and supported. Thus, I tried DevCore31, but that failed, too. The reason for that is the conditional markup below, which uses `DevCore30` instead of `DevCore31`.

```
    <When Condition=" '$(ProjectLoadStyle)' == 'DevCore30' ">
      <PropertyGroup>
        <ProductTargetFrameworks>netstandard2.0</ProductTargetFrameworks>
        <TestTargetFrameworks>netcoreapp3.1</TestTargetFrameworks>
        <AssetsTargetFrameworks>netstandard1.3</AssetsTargetFrameworks>
        <BenchmarkTargetFrameworks>netcoreapp3.1</BenchmarkTargetFrameworks>
        <SamplesFrameworks>netcoreapp3.1</SamplesFrameworks>
      </PropertyGroup>
    </When>
```

Next, I removed the `ProjectLoadStyle` environment variable and, voila, it worked and built the default, which is `DevCore21`.

To sum up, there are indeed issues that should be fixed:

- The build instructions are silent on the `ProjectLoadStyle` environment variable (and potentially other things as well). They should thus be amended. I had totally forgotten that I once defined that environment variable. It would have been helpful to have build instructions that identify the things that can be configured and might have become invalid.
- The `Directory.Build.props` configuration file should be fixed.
